addons/delivery/models/stock_picking.py

```diff
@@ -230,10 +230,11 @@ def send_to_shipper(self):
         self.carrier_price = res['exact_price'] * (1.0 + (self.carrier_id.margin / 100.0))
         if res['tracking_number']:
             previous_pickings = self.env['stock.picking']
-            previous_moves = self.move_lines.move_orig_ids
+            accessed_moves = previous_moves = self.move_lines.move_orig_ids
             while previous_moves:
                 previous_pickings |= previous_moves.picking_id
-                previous_moves = previous_moves.move_orig_ids
+                previous_moves = previous_moves.move_orig_ids.filtered(lambda move: move not in accessed_moves)
```

Comment:
A bit more performant (uses a set behind the scene) and readable IMO
```suggestion
                previous_moves = previous_moves.move_orig_ids - accessed_moves
```

Reply:
Jep, sometimes you miss the most obvious way of doing things...accepted and committed 😄